modules/thermalmanager.c

```diff
@@ -52,545 +156,995 @@
 
-static GSList* thermal_objects = 0;
+static void thermal_manager_get_thermal_state_cb       (const DsmeDbusMessage *request, DsmeDbusMessage **reply);
 
-static const char* const service   = thermalmanager_service;
-static const char* const interface = thermalmanager_interface;
-static const char* const path      = thermalmanager_path;
+static void thermal_manager_handle_temperature_query   (const DsmeDbusMessage *req, const char *sensor_name, DsmeDbusMessage **rsp);
+static void thermal_manager_get_surface_temperature_cb (const DsmeDbusMessage *req, DsmeDbusMessage **rsp);
+static void thermal_manager_get_core_temperature_cb    (const DsmeDbusMessage *req, DsmeDbusMessage **rsp);
+static void thermal_manager_get_battery_temperature_cb (const DsmeDbusMessage *req, DsmeDbusMessage **rsp);
+static void thermal_manager_get_sensor_temperature_cb  (const DsmeDbusMessage *req, DsmeDbusMessage **rsp);
 
+/* ------------------------------------------------------------------------- *
+ * MODULE_GLUE
+ * ------------------------------------------------------------------------- */
+
+void module_init (module_t *handle);
+void module_fini (void);
+
+/* ========================================================================= *
+ * MODULE_DATA
+ * ========================================================================= */
+
+/** DSME module handle for this module */
+static module_t *this_module = 0;
+
+/** List of registered thermal objects */
+static GSList *thermal_objects = 0;
+
+/** Currently accepted device thermal state */
 static THERMAL_STATUS current_status = THERMAL_STATUS_NORMAL;
 
-#ifdef DSME_THERMAL_TUNING
-static bool is_in_ta_test = false;
-#endif
+/** Flag for: D-Bus method handlers have been registered */
+static bool dbus_methods_bound = false;
 
-static const char* const thermal_status_name[] = {
-  "low-temp-warning", "normal", "warning", "alert", "fatal"
-};
+/** D-Bus name of the thermal manager service */
+static const char * const dbus_service   = thermalmanager_service;
 
+/** Thermal manager D-Bus interface */
+static const char * const dbus_interface = thermalmanager_interface;
 
-static const char* current_status_name()
+/** Thermal manager D-Bus object path */
+static const char * const dbus_object    = thermalmanager_path;
+
+/* ========================================================================= *
+ * THERMAL_STATUS
+ * ========================================================================= */
+
+/** Convert thermal status to name used in D-Bus signaling
+ *
+ * @param  status thermal status
+ *
+ * @return string expected by dbus peers
+ */
+const char *
+thermal_status_name(THERMAL_STATUS status)
 {
-  return thermal_status_name[current_status];
+    /* Note: Any changes to the strings returned by this function
+     *       constitute D-Bus API break and thus should be avoided.
+     */
+
+    const char *res = "unknown";
+
+    switch( status ) {
+    case THERMAL_STATUS_LOW:     res = "low-temp-warning"; break;
```

Comment:
These strings here should probably appear in https://github.com/nemomobile/libdsme/blob/master/modules/thermalmanager_dbus_if.h because right now for e.g. lipstick they are hardcoded in qmsystem2, in `qmthermal_p.h`
as `#define LOW_TEMP_WARNING "low-temp-warning"`. To be in line with the rest of `thermalmanager_dbus_if.h`, I'd suggest adding to it:

```
extern const char thermalmanager_thermal_status_low[];
extern const char thermalmanager_thermal_status_normal[];
extern const char thermalmanager_thermal_status_warning[];
extern const char thermalmanager_thermal_status_alert[];
extern const char thermalmanager_thermal_status_fatal[];
```

(and adding the string definitions to `thermalmanager_dbus_if.c`)

This way, we can get rid of literal strings here and in the qmthermal code parts and use the common strings. This then needs to link to `-lthermalmanager_dbus_if` if it doesn't do that already.